src/checkers/inference/DefaultSlotManager.java

```diff
@@ -359,6 +374,25 @@ public ExistentialVariableSlot createExistentialVariableSlot(VariableSlot potent
         return existentialVariableSlot;
     }
 
+    @Override
+    public ArithmeticVariableSlot createArithmeticVariableSlot(ArithmeticOperationKind operation,
+            VariableSlot leftOperand, VariableSlot rightOperand) {
+        ArithmeticVariableSlot arithmeticVariableSlot;
+        Pair<Slot, Slot> pair = new Pair<>(leftOperand, rightOperand);
+
+        Map<Pair<Slot, Slot>, Integer> slotsForOp = arithmeticSlotCache.get(operation);
+        if (slotsForOp.containsKey(pair)) {
+            int id = slotsForOp.get(pair);
+            arithmeticVariableSlot = (ArithmeticVariableSlot) getVariable(id);
+        } else {
+            arithmeticVariableSlot = new ArithmeticVariableSlot(
+                    null, nextId(), operation, leftOperand, rightOperand);
```

Comment:
I started this method based on combine constraint, will add it as a parameter and pass it in.

Reply:
`CombineConstraint` passed null because not all the clients can pass correct locations. But for arithmetic constraint, I think it's generated only on binary trees and unary trees(desugar `i++` to i = i + 1)? Then locations may be more clear than CombineConstraint. So passing valid location would be duable, I guess.